test/Faker/Provider/nl_NL/PersonTest.php

```diff
@@ -21,6 +21,71 @@ public function testGenerateValidIdNumber()
         self::assertTrue($sum != 0 && $sum % 11 == 0);
     }
 
+    /**
+     * @dataProvider identityDocumentNumberProvider
+     * @param \DateTime   $issueDate
+     * @param string|null $documentCode
+     * @param string      $regex
+     */
+    public function testGenerateValidIdentityDocumentNumber(\DateTime $issueDate, ?string $documentCode, string $regex)
+    {
+        self::assertMatchesRegularExpression($regex, $this->faker->identityDocumentNumber($issueDate, $documentCode));
+    }
+
+
+    /**
+     * @dataProvider validityProvider
+     * @param int $age
+     * @param int $validity
+     */
+    public function testGenerateValidIdentityDocumentValidity(int $age, int $validity)
+    {
+        self::assertSame($validity, $this->faker->identityDocumentValidity($age));
+    }
+
+    /**
+     * @return array
+     */
+    public function identityDocumentNumberProvider()
+    {
+        return [
+            [new \DateTime('2019-11-30'), null, '/^[INDSARB][A-Z][A-Z0-9]{6}[0-9]{1}$/'],
+            [new \DateTime('2019-12-01'), null, '/^[INDSARB][A-Z][A-Z1-9]{6}[0-9]{1}$/'],
```

Comment:
And in every second these no 1 is allowed, e.g. here
`/^[INDSARB][A-NP-Z][A-NP-Z1-9]{6}[1-9]{1}$/`

Reply:
This is because `// No '0' in documents issued from 1 december 2019`. But I corrected a typo indeed in the second `0-9`.